exporter/opentelemetry-exporter-zipkin/src/opentelemetry/exporter/zipkin/__init__.py

```diff
@@ -116,6 +121,11 @@ def __init__(
             self.encoder = ProtobufEncoder(max_tag_value_length)
 
     def export(self, spans: Sequence[Span]) -> SpanExportResult:
+        # Populate service_name from first span
+        if spans:
+            service_name = spans[0].resource.attributes.get(SERVICE_NAME)
+            if service_name:
+                self.local_node.service_name = service_name
```

Comment:
Initially I was gonna recommend to move this logic to encoder but then I remembered we decided to restrict using one span processor with only a single provider so this is good. May be we should add a comment explaining this here so others don't think this is a bug?
|